core/embed/sec/optiga/optiga.c

```diff
@@ -115,6 +115,47 @@ static const optiga_metadata_item ACCESS_PIN_HMAC_CTR =
 // Size of the CMAC/HMAC prefix returned by Optiga.
 #define ENCRYPT_SYM_PREFIX_SIZE 3
 
+static uint32_t sec_clr_time_s = 0;  // Estimated time needed to clear SEC
+
+optiga_result optiga_init() {
+  sec_clr_time_s = 0;
+  return optiga_transport_init();
+}
+
+void optiga_deinit() {
+  sec_clr_time_s = 0;
+  optiga_transport_deinit(false);
+}
+
+void optiga_soft_deinit() {
+  uint8_t sec;
+  bool status = optiga_read_sec(&sec);
+
+  // if SEC counter is above the threshold, deinitialize the transport layer,
+  // but do not power down the chip yet, so the SEC counter could decrement
+  if (status && sec > OPTIGA_SEC_SUSPEND_THR) {
+    optiga_transport_deinit(true);
+    sec_clr_time_s = sec * (OPTIGA_T_MAX_MS / 1000);
```

Comment:
No reason to divide before multiplying. It's better the other way around. If `OPTIGA_T_MAX_MS` is changed to 900, for example, then this code will always result in 0, whereas dividing after multiplying will give the correct result.
```suggestion
    sec_clr_time_s = (sec * OPTIGA_T_MAX_MS) / 1000;
```